lib/buffer.js

```diff
@@ -514,10 +514,8 @@ Buffer.prototype.fill = function fill(val, start, end) {
   start = start >> 0;
   end = (end === undefined) ? this.length : end >> 0;
 
-  if (start < 0 || end > this.length)
+  if (start < 0 || end > this.length || start >= this.length)
```

Comment:
It should have been `start >= end`. Even if `end < this.length` and `start >= end`, it still should be an error.